modules/mauth-common/src/main/java/com/mdsol/mauth/MAuthRequest.java

```diff
@@ -139,9 +139,8 @@ private MAuthRequest(String authenticationHeaderValue, byte[] messagePayload, In
 
     // we always use inputStream for Auth,
     // wrap byte array in ByteArrayInputStream if bodyInputStream doesn't set
-    if (bodyInputStream == null) {
-      if (messagePayload == null) messagePayload = new byte[0];
-      this.bodyInputStream = new ByteArrayInputStream(messagePayload);
+    if (bodyInputStream == null && messagePayload != null) {
+       this.bodyInputStream = new ByteArrayInputStream(messagePayload);
     }
```

Comment:
throw an exception in the else clause. Something is really wrong if the `if` branch did not match

Reply:
updated in https://github.com/mdsol/mauth-jvm-clients/pull/132/commits/16a30f1c09fed3a1590ed3b3e8cd106a367ce027